worlds/oot_soh/location_access/overworld/kakariko.py

```diff
@@ -7,6 +7,356 @@
+    connect_regions(Regions.KAK_SHOOTING_GALLERY, world, [
+        (Regions.KAKARIKO_VILLAGE, lambda bundle: True),
+    ])
+
+    # Kak Potion Shop Front
+    # Locations
+    add_locations(Regions.KAK_POTION_SHOP_FRONT, world, [
+        (Locations.KAK_POTION_SHOP_ITEM1, lambda bundle: is_adult(bundle)),
+        (Locations.KAK_POTION_SHOP_ITEM2, lambda bundle: is_adult(bundle)),
+        (Locations.KAK_POTION_SHOP_ITEM3, lambda bundle: is_adult(bundle)),
+        (Locations.KAK_POTION_SHOP_ITEM4, lambda bundle: is_adult(bundle)),
+        (Locations.KAK_POTION_SHOP_ITEM5, lambda bundle: is_adult(bundle)),
+        (Locations.KAK_POTION_SHOP_ITEM6, lambda bundle: is_adult(bundle)),
+        (Locations.KAK_POTION_SHOP_ITEM7, lambda bundle: is_adult(bundle)),
+        (Locations.KAK_POTION_SHOP_ITEM8, lambda bundle: is_adult(bundle)),
+    ])
+    # Connections
+    connect_regions(Regions.KAK_POTION_SHOP_FRONT, world, [
+        (Regions.KAKARIKO_VILLAGE, lambda bundle: True),
+        (Regions.KAK_POTION_SHOP_BACK, lambda bundle: is_adult(bundle)),
+    ])
+
+    # Kak Potion Shop Back
+    # Connections
+    connect_regions(Regions.KAK_POTION_SHOP_BACK, world, [
+        (Regions.KAK_POTION_SHOP_FRONT, lambda bundle: True),
+        (Regions.KAK_BACKYARD, lambda bundle: is_adult(bundle)),
+    ])
+
+    # Kak Granny's Potion Shop
+    # Locations
+    add_locations(Regions.KAK_GRANNYS_POTION_SHOP, world, [
+        [Locations.KAK_TRADE_ODD_MUSHROOM,
+         lambda bundle: is_adult(bundle) and can_use(Items.ODD_MUSHROOM, bundle)],
+        (Locations.KAK_GRANNYS_SHOP,
+         lambda bundle: is_adult(bundle) and (can_use(Items.ODD_MUSHROOM, bundle)) and trade_quest_step(
+             Items.ODD_MUSHROOM, bundle))
+    ])
+    # Connections
+    connect_regions(Regions.KAK_GRANNYS_POTION_SHOP, world, [
+        (Regions.KAK_BACKYARD, lambda bundle: True)
+    ])
+
+    # Kak Redead Grotto
+    # Locations
+    add_locations(Regions.KAK_REDEAD_GROTTO, world, [
+        (Locations.KAK_REDEAD_GROTTO_CHEST,
+         lambda bundle: can_kill_enemy(bundle, enemy=Enemies.REDEAD, combat_range=EnemyDistance.CLOSE,
+                                       wall_or_floor=True, quantity=2))
+    ])
+    # Connections
+    connect_regions(Regions.KAK_REDEAD_GROTTO, world, [
+        (Regions.KAKARIKO_VILLAGE, lambda bundle: True),
+    ])
+
+    # Kak Open Grotto
+    # Locations
+    add_locations(Regions.KAK_OPEN_GROTTO, world, [
+        (Locations.KAK_OPEN_GROTTO_CHEST, lambda bundle: True),
+        (Locations.KAK_OPEN_GROTTO_FISH, lambda bundle: has_bottle(bundle)),
+        (Locations.KAK_OPEN_GROTTO_GOSSIP_STONE_FAIRY, lambda bundle: call_gossip_fairy(bundle)),
+        (Locations.KAK_OPEN_GROTTO_GOSSIP_STONE_BIG_FAIRY, lambda bundle: can_play_song(bundle, Items.SONG_OF_STORMS)),
+        (Locations.KAK_OPEN_GROTTO_GOSSIP_STONE, lambda bundle: True),
+        (Locations.KAK_OPEN_GROTTO_BEEHIVE_LEFT, lambda bundle: can_break_lower_hives(bundle)),
```

Comment:
`KAK_OPEN_GROTTO_GOSSIP_STONE` is only used as a hint location in Ship, it serves no purpose for us. Feel free to remove it.